docs/migrate-to-antrea.md

```diff
@@ -0,0 +1,147 @@
+# Migrate from other CNI to Antrea
+
+This document describes how to migrate from other CNIs to Antrea. Currently, we support migration
+from Calico and Flannel.
+As for Calico, NetworkPolicy conversion is supported conditionally. As for Flannel, only Pod
+networking migration is supported.
+
+NOTE: The migration process is only verified on Kubernetes v1.27.3, Calico v3.26 and Flannel
+v0.22.3. Other versions may not be supported. And the migration process only supports
+Linux Node.
+
+With the help of migration [script](../hack/migrate-to-antrea.sh) and the `antctl migrate` command,
+the migration process is fully automated. You can use one of the following commands to start the migration:
+
+```bash
+# Migrate from Calico to Antrea
+./hack/migrate-to-antrea.sh --from calico
+# Migrate from Flannel to Antrea
+./hack/migrate-to-antrea.sh --from flannel
+```
+
+More details about the migration process are described in the following sections. And you can learn how to use
+the script by running `./hack/migrate-to-antrea.sh --help`.
+
+The migration process is divided into four steps:
+
+1. Install Antrea in the cluster.
+2. If necessary, convert NetworkPolices from other CNI to Antrea.
```

Comment:
Is this the only reason why we need to specify `--from` in the command? If yes, I feel it's not worth to do the conversion in Antrea itself, as it needs too many knowledge to do this job and may need continuous support, and it's not guaranteed that there is always a 1:1 mapping, there could also be version issues.
cc @jianjuns @antoninbas for opinions.

Reply:
Yes, I mean not to convert any CNI provider specific NetworkPolicy here. It would also make the tool more generic. If someone really needs to covert some CNI provider specific resource in the future, it doesn't really need to be part of this migrator, a 3rd-party tool may be more proper.
The scope of the migrator in my mind:
Should not do:
1. uninstallaton of the original CNI (too many cases/method to support, and perhaps look aggressive)
2. resource conversion of the original CNI specific resources (ditto)

Should do:
1. migrate Pods network to Antrea CNI
2. cleanup of stale network resources/configurations on the system that may affect the Pod's network after migration

Reply:
Sounds good to me.

For this one:
> cleanup of stale network resources/configurations on the system that may affect the Pod's network after migration

can this be done in a generic way?

Reply:
> Should not do:
uninstallaton of the original CNI (too many cases/method to support, and perhaps look aggressive, it's not hard for users to uninstall anyway)
resource conversion of the original CNI specific resources (ditto)

Hi Quan. I am not sure whether the Network will work properly since there may be some route conflict. Maybe we just need to output a message to reminder users to uninstall their old CNIs.?

Reply:
> > cleanup of stale network resources/configurations on the system that may affect the Pod's network after migration
> 
> can this be done in a generic way?

I suppose no because every CNI has its own configuration on the system while it's too risky to just remove everything that are not needed by Antrea. But I think there aren't too many cleanup: iptables(/nftables), ip link, ip route.

> Hi Quan. I am not sure whether the Network will work properly since there may be some route conflict. Maybe we just need to output a message to reminder users to uninstall their old CNIs.?

"cleanup of stale network resources/configurations on the system" is part of "Should do", do I misunderstand your question?

I think uninstalling old CNIs should be done before the migration, otherwise old CNIs can create their resources back after cleanup. I'm fine with outputing a message to remind users, perhaps it could be part of precheck.

@antoninbas @hjiajing furthermore, I'm thinking if we should just let users uninstall old CNI and install Antrea in anyway they like, then run the migrator which takes care of migrating existing Pods and cleanup of stale network resources. This should make the migrator simpler and easier to run it in various distributions.

Reply:
> I'm thinking if we should just let users uninstall old CNI and install Antrea in anyway they like, then run the migrator which takes care of migrating existing Pods and cleanup of stale network resources. This should make the migrator simpler and easier to run it in various distributions

Works for me

Reply:
Removed the NP conversion and uninstall old CNI steps.